app/scripts/controllers/permissions/index.js

```diff
@@ -716,7 +724,7 @@ export class PermissionsController {
        * @param {string} req - The internal rpc-cap user request object.
        */
       requestUserApproval: async (req) => {
-        const { origin, metadata: { id } } = req
+        const { metadata: { id, origin } } = req
```

Comment:
Isn't the `id` part of the `req` object now? Not the metadata
```suggestion
        const { id, metadata: { origin } } = req
```

Reply:
No, it's in the `metadata`: https://github.com/MetaMask/rpc-cap/blob/master/src/@types/index.d.ts/#L23-L31

Reply:
Ah, I see. Different metadata object.